### PR TITLE
Remove unnecessary attribute in Instrument

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Removed `labels` attribute, as it is not needed anymore.  This addresses #483 

## Type of change

Please delete options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)

# How Has This Been Tested?

Ran unit tests locally

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### Downstream modules:
- [x] pysatSpaceWeather: https://github.com/pysat/pysatSpaceWeather/pull/32
- [x] pysatMissions: @jklenzing 
- [x] pysatNASA: https://github.com/pysat/pysatNASA/pull/25
- [x] pysatIncubator: @rstoneback 
- [x] pysatModels: https://github.com/pysat/pysatModels/pull/69
- [x] pysatMadrigal: https://github.com/pysat/pysatMadrigal/pull/28
- [x] pysatCDAAC: pysat/pysatCDAAC#11

### CODE REVIEW

**Comment by aburrell** [comment]
I don't believe any changes to Missions or Incubator are necessary, but would like a second opinion.

File: pysat/_instrument.py [hunk 1]
```diff
@@ -2004,14 +2000,8 @@ def custom_apply_all(self):
                         recast_as_dict = False
                         if not isinstance(new_data, dict):
                             new_data = {'data': new_data}
-                            # dstr = ''.join(('Data returned should be a dict ',
```
Comment (rstoneback) [inline pysat/_instrument.py:2007]: This didn't last long. Technically you just approved it the other day :P

After our discussion on Tue. this deprecation warning will be effectively moving to the custom_attach for anything other than 'modify'.

**Review by rstoneback — APPROVED** [review]

File: pysat/_instrument.py (hunk 1 above)
Comment (aburrell) [reply pysat/_instrument.py:2007]: Yeah, but then we had our discussion!  We've gotta stay on our toes :)

**Comment by rstoneback** [comment]
I checked pysatIncubator. I found a couple uses of meta.labels, which is in spec, and no uses of pysat.Instrument.labels.

**Comment by rstoneback** [comment]
pysatMissions also looked good to me. meta.labels was used but not the .labels being removed here.

**Pull request merged**